tests/helpers/test_device_registry.py

```diff
@@ -0,0 +1,78 @@
+"""Tests for the Device Registry."""
+import pytest
+
+from homeassistant.helpers import device_registry
+
+
+def mock_registry(hass, mock_entries=None):
+    """Mock the Device Registry."""
+    registry = device_registry.DeviceRegistry(hass)
+    registry.devices = mock_entries or []
+
+    async def _get_reg():
+        return registry
+
+    hass.data[device_registry.DATA_REGISTRY] = \
+        hass.loop.create_task(_get_reg())
+    return registry
+
+
+@pytest.fixture
+def registry(hass):
+    """Return an empty, loaded, registry."""
+    return mock_registry(hass)
+
+
+async def test_get_or_create_returns_same_entry(registry):
+    """Make sure we do not duplicate entries."""
+    entry = registry.async_get_or_create(
+        [['bridgeid', '0123']], 'Dresden Elektronik', 'Conbee',
+        [['Ethernet', '1.0.0.0']])
```

Comment:
IP addresses should never be used for connectivity.

Ethernet should also just be mac address, as we might just now if the other party is on our network, not perse if they are wireless / connected. It should also be a constant in the device_registry.py file.

Reply:
And probably too be lower case.

Reply:
I'll make it a better example